lib/cache/ExpirationStrategy.ts

```diff
@@ -0,0 +1,40 @@
+import { StorageType } from './StorageType';
+import { MemoryStorage } from './MemoryStorage';
+
+export class ExpirationStrategy<K> {
+
+  private readonly expirations = new MemoryStorage<K>();
+
+  constructor(
+    private readonly storage: StorageType<K>,
+    private readonly expiration: 'absolute' | 'sliding',
+    private readonly timeout: number,
+  ) { }
+
+  public add(key: K): void {
+    this.addKey(key);
+  }
+
+  public update(key: K): void {
```

Comment:
`update` name is fine. But I think we could use better the name `touch`.
The idea is that we only touch the value, not update it.